client/app/scripts/liveblog-edit/pages-manager.service.js

```diff
@@ -221,21 +221,16 @@
                         addPage(page);
                         page = undefined;
                     }
-                    angular.forEach(post.items, function(item) {
-                        if (item.item.item_type === 'embed') {
-                            if (item.item.text.indexOf('platform.instagram.com') !== -1) {
-                                processInstagram = true;
-                            }
-                        }
-                    });
+
+                    if (instagramService.postHasEmbed(post.items)) {
```

Comment:
assignment is much better here aka
processInstagram = instagramService.postHasEmbed(post.items);